EIPS/eip-00.md

```diff
@@ -0,0 +1,831 @@
+operating on L2 rollups, and given that this work does not correspond to the amount of gas spent on
+validation and is not linked to the gas price, the `sender` may decide
+to pay an extra `builderFee` as a "tip" to the block builder.
+
+This value is denominated in wei and is passed from the `sender` to the `coinbase` as part of the gas pre-charge.
+
+### Multiple execution frames for a single transaction
+
+All existing transaction types only have an implicit validation phase where balance, nonce, and signature are checked,
+and a single top-level execution frame with
+`tx.origin == msg.sender` which is the address that is determined by a transaction ECDSA signature.
+
+When processing a transaction of type `AA_TX_TYPE`, however, multiple execution frames will be created.
+The full list of possible frames tries to replicate the ERC-4337 flow:
+
+1. Validation Phase
+   * `nonce` validation and increment frame (required)
+   * `sender` deployment frame (once per account)
+   * `sender` validation frame (required)
+   * `paymaster` validation frame (optional)
+2. Execution Phase
+   * `sender` execution frame (required)
+   * `paymaster` post-transaction frame (optional)
+
+All execution frames in the "Validation Phase" must be completed successfully without reverting, and the return value
+for `sender` and `paymaster` validation frames must include `MAGIC_VALUE_SENDER` and `MAGIC_VALUE_PAYMASTER` accrodingly
+in order for the transaction to be considered valid for a given position in a block.
+
+In terms of block validity, all validation and execution frames may read and write any state when included in the block.
+However, the AA transactions in the mempool SHOULD be bound by storage access rules to avoid DoS on block builders.
+These rules are defined in EIP-9999.
+
+In all top-level frames, the global variables have the following meaning:
+
+| Opcode Name | Solidity Equivalent | Value                                                                         |
+|-------------|---------------------|-------------------------------------------------------------------------------|
+| `CALLER`    | `msg.sender`        | The `AA_ENTRY_POINT` address. `AA_SENDER_CREATOR` for the "deployment frame". |
+| `ORIGIN`    | `tx.origin`         | The transaction `sender` address                                              |
+| `CALLDATA*` | `msg.data`          | The transaction data is set to inputs of the corresponding frame              |
+
+#### Nonce validation frame
+
+The `NonceManager` is invoked with the following data:
+
+```solidity
+abi.encodePacked(sender, nonce)
+```
+
+#### Sender deployment frame
+
+The `deployer` address is invoked with the `deployerData[20:]` as call data input.
+It is important that the `deployer` is **not** invoked from the `AA_ENTRY_POINT` but from the `AA_SENDER_CREATOR`.
+This is necessary to guarantee that `AA_ENTRY_POINT` may never initiate a call to a `sender` execution function
```

Comment:
The issue is that `AA_ENTRY_POINT` is trusted by the account to perform arbitrary calls, and the protocol guarantees that an account would only receive such calls after successful validation.  Calling a deployer with arbitrary calldata from `AA_ENTRY_POINT` breaks that assumption.

Consider a scenario where the victim uses an account that supports batching via an `executeBatch(calls[])` function, and an attacker wants to take over the account.

The attacker would send a transaction from a new sender (with no code) and deployerData pointing to the victim's account.  The deployer data will be a call to `executeBatch` where calls[0] is a call to the real factory which will deploy the sender.  And calls[1] is a `victim.setOwner` or some other privileged function in the victim's account.

The transaction will be valid because it properly deploys the `sender`, and then the sender approves the validity of the transaction.  But it violated the victim's expectation, causing the account's `executeBatch` to be called by AA_ENTRY_POINT without passing validation.

Reply:
Got it, thank you